## Exercise 1

In [3]:
def hRatio(r1, p1, r2, p2):
    return (r2**2 / r1**2) * (p1 / p2)

In [5]:
sunJupiter = hRatio(7e8, 2.4e6, 7.5e11, 3.6e8)
sunNeptune = hRatio(7e8, 2.4e6, 4.5e12, 5.2e9)
jupiterIo = hRatio(7e7, 3.5e4, 4.2e8, 1.5e5)
jupiterCallisto = hRatio(7e7, 3.5e4, 1.8e9, 1.4e6)
saturnTitan = hRatio(5.8e7, 3.6e4, 1.2e9, 1.3e6)
uranusOberon = hRatio(2.8e7, 6.2e4, 5.8e8, 1.2e6)

print("Sun/Jupiter: %d" % sunJupiter)
print("Sun/Neptune: %d" % sunNeptune)
print("Jupiter/Io: %d" % jupiterIo)
print("Jupiter/Callisto: %d" % jupiterCallisto)
print("Saturn/Titan: %d" % saturnTitan)
print("Uranus/Oberon: %d" % uranusOberon)

Sun/Jupiter: 7653
Sun/Neptune: 19073
Jupiter/Io: 8
Jupiter/Callisto: 16
Saturn/Titan: 11
Uranus/Oberon: 22


The extreme difference between the ratios highlight the peculiarity of angular momentum in the sun - planet systems. The sun is spinning far too slowly given the angular momentum of the planets. It captured most of the mass in the solar system, yet the planets have most of the angular momentum. The planet - moon systems make more sense, since the relatively massive planet has greater angular momentum than its moons.

## Exercise 2

In [31]:
def Δr1(r1, r2, Δr2):
    return -(r1 / r2)**(1/2) * Δr2

In [51]:
# Constants
G = 6.67408e-11  # Gravitational constant
sun = 1.989e30   # Mass of Sun (kg)
jup = 1.898e27   # Mass of Jupiter (kg)

# Current semimajor axes (AU)
jupCur = 5.2028
satCur = 9.5388
urnCur = 19.1914
nepCur = 30.0611

# Early semimajor axes (AU)
jupEarly = 7.5
satEarly = 9
urnEarly = 12
nepEarly = 15

# Change in jupiter's orbit
jupSat = Δr1(jupCur, satCur, satCur - satEarly)
jupUrn = Δr1(jupCur, urnCur, urnCur - urnEarly)
jupNep = Δr1(jupCur, nepCur, nepCur - nepEarly)
jupTot = jupSat + jupUrn + jupNep

print("Jupiter's migration: %.2f AU" % jupTot)

# Jupiter's loss of energy
L1 = (G * sun * jupCur)**(1/2) * jup
L2 = (G * sun * jupEarly)**(1/2) * jup

print("Jupiter's energy loss: %.2E kg-m^2/s" % (L1 - L2))

Jupiter's migration: -10.41 AU
Jupiter's energy loss: -1.00E+37 kg-m^2/s


I initially thought a 10.41 AU migration inward was absurdly large, but further research showed that Jupiter formed between 3 and 10 AU before migrating inwards to about 1.5 AU during the "Grand Tack". 10.41 AU is still a little big, but not unreasonably so if we're looking at the higher bound.

https://www.nasa.gov/topics/solarsystem/features/young-jupiter.html - Grand Tack
http://www.pnas.org/content/pnas/112/14/4214.full.pdf               - 3 to 10 AU range